modules/yardstick/src/main/java/org/apache/ignite/yardstick/sql/IgniteInlineIndexBenchmark.java

```diff
@@ -110,16 +145,29 @@ private void init() {
 
     /** Creates next random key. */
     private Object nextKey() {
-        return isJavaObj
-            ? new TestKey(ThreadLocalRandom.current().nextInt(range))
-            : ThreadLocalRandom.current().nextInt(range);
+        switch (testedType) {
+            case JAVA_OBJECT:
+                return new TestKey(ThreadLocalRandom.current().nextInt(range));
+            case PRIMITIVE:
+                return ThreadLocalRandom.current().nextInt(range);
+            case DECIMAL:
+                return BigDecimal.valueOf(ThreadLocalRandom.current().nextDouble(range));
```

Comment:
we should limit amount of different keys, not its upper bound. Otherwise the test will fail from time to time since memory limit is set for default data region